qt/widgets/common/test/FitDomainTest.h

```diff
@@ -0,0 +1,544 @@
+    m_fitDomain->setFunction(m_flatBackground);
+
+    m_fitDomain->setParameterValue("A0", 3.0);
+
+    TS_ASSERT_EQUALS(m_fitDomain->getParameterValue("A0"), 3.0);
+  }
+
+  void test_that_getAttributeValue_will_get_the_attribute_value_if_it_exists() {
+    m_fitDomain->setFunction(m_composite);
+    TS_ASSERT(!m_fitDomain->getAttributeValue("NumDeriv").asBool());
+  }
+
+  void
+  test_that_getAttributeValue_will_throw_if_the_stored_function_is_a_nullptr() {
+    TS_ASSERT_THROWS(m_fitDomain->getAttributeValue("A0"),
+                     std::runtime_error const &);
+  }
+
+  void
+  test_that_getAttributeValue_will_throw_if_the_attribute_does_not_exist() {
+    m_fitDomain->setFunction(m_flatBackground);
+    TS_ASSERT_THROWS(m_fitDomain->getAttributeValue("Height"),
+                     std::runtime_error const &);
+  }
+
+  void
+  test_that_setAttributeValue_will_not_throw_if_the_stored_function_is_a_nullptr() {
+    Mantid::API::IFunction::Attribute value(true);
+    TS_ASSERT_THROWS_NOTHING(m_fitDomain->setAttributeValue("NumDeriv", value));
+  }
+
+  void
+  test_that_setAttributeValue_will_not_throw_if_the_stored_function_does_not_have_the_specified_attribute() {
+    Mantid::API::IFunction::Attribute value(true);
+    m_fitDomain->setFunction(m_flatBackground);
+
+    TS_ASSERT_THROWS_NOTHING(m_fitDomain->setAttributeValue("NumDeriv", value));
+  }
+
+  void
+  test_that_setAttributeValue_will_set_the_attribute_value_ok_if_it_is_valid() {
+    Mantid::API::IFunction::Attribute value(true);
+    m_fitDomain->setFunction(m_composite);
+
+    m_fitDomain->setAttributeValue("NumDeriv", value);
+
+    TS_ASSERT(m_fitDomain->getAttributeValue("NumDeriv").asBool());
+  }
+
+  void
+  test_that_hasParameter_returns_false_if_the_stored_function_is_a_nullptr() {
+    TS_ASSERT(!m_fitDomain->hasParameter("A0"));
+  }
+
+  void
+  test_that_hasParameter_returns_false_if_the_function_does_not_have_a_parameter() {
+    m_fitDomain->setFunction(m_flatBackground);
+    TS_ASSERT(!m_fitDomain->hasParameter("Height"));
+  }
+
+  void
+  test_that_hasParameter_returns_false_if_the_function_does_have_a_parameter() {
+    m_fitDomain->setFunction(m_flatBackground);
+    TS_ASSERT(m_fitDomain->hasParameter("A0"));
+  }
+
+  void
+  test_that_isParameterActive_returns_false_if_the_stored_function_is_a_nullptr() {
+    TS_ASSERT(!m_fitDomain->isParameterActive("A0"));
+  }
+
+  void
+  test_that_isParameterActive_returns_false_if_the_function_does_not_have_the_specified_parameter() {
+    m_fitDomain->setFunction(m_flatBackground);
+    TS_ASSERT(!m_fitDomain->isParameterActive("Height"));
+  }
+
+  void test_that_isParameterActive_returns_false_if_a_parameter_is_tied() {
+    m_fitDomain->setFunction(m_composite);
+    TS_ASSERT(m_fitDomain->updateParameterTie("f0.A0", "f1.Height"));
+
+    TS_ASSERT(!m_fitDomain->isParameterActive("f0.A0"));
+    TS_ASSERT(m_fitDomain->isParameterActive("f1.Height"));
```

Comment:
is there a reason why height is the active parameter and not A0?

Reply:
Yes, height is active because it isn't tied to another parameter. However A0 has a tie to Height.